unittests/security_group_tests.cpp

```diff
@@ -0,0 +1,396 @@
+   (local $bytes_remaining i32)
+   (local $required_bytes i32)
+   (set_local $bytes_remaining (call $action_data_size))
+   (drop (call $read_action_data (i32.const 0) (get_local $bytes_remaining)))
+   (set_local $required_bytes (call $get_active_security_group (i32.const 0) (i32.const 0)))
+   (call $eosio_assert (i32.eq (i32.const 13) (get_local $required_bytes))(i32.const 512))
+   (drop (call $get_active_security_group (i32.const 256) (get_local $required_bytes)))
+   (call $eosio_assert (i32.eq (call $memcmp (i32.const 0) (i32.const 256) (get_local $required_bytes))(i32.const 0)) (i32.const 768)) 
+ )
+ (data (i32.const 512) "get_active_security_group should return the right size")
+ (data (i32.const 768) "get_active_security_group output buffer must match the input")
+)
+)=====";
+
+std::vector<char> participants_payload( participants_t names ) {
+   fc::datastream<std::vector<char>> ds;
+   fc::raw::pack(ds, names);
+   return ds.storage();
+}
+
+BOOST_AUTO_TEST_CASE(test_security_group_intrinsic) {
+
+   eosio::testing::tester chain1;
+   using namespace eosio::chain::literals;
+
+   chain1.create_accounts( {"alice"_n,"bob"_n,"charlie"_n} );
+   chain1.produce_blocks(3);
+
+   chain1.create_accounts({ "addmember"_n, "rmmember"_n, "ingroup"_n, "getgroup"_n });
+   chain1.produce_block();
+
+   chain1.set_code( "addmember"_n, propose_security_group_participants_add_wast );
+   chain1.set_code( "rmmember"_n, propose_security_group_participants_remove_wast );
+   chain1.set_code( "ingroup"_n, assert_in_security_group_wast );
+   chain1.set_code( "getgroup"_n, assert_get_security_group_wast );
+   chain1.produce_block();
+   
+   chain1.push_action( "eosio"_n, "setpriv"_n, "eosio"_n, fc::mutable_variant_object()("account", "addmember"_n)("is_priv", 1));
+   chain1.push_action( "eosio"_n, "setpriv"_n, "eosio"_n, fc::mutable_variant_object()("account", "rmmember"_n)("is_priv", 1));
+   chain1.produce_blocks(24);
+
+   chain1.set_producers( {"alice"_n,"bob"_n} );
+   chain1.produce_blocks(3); // Starts new blocks which promotes the proposed schedule to pending
+   BOOST_REQUIRE_EQUAL( chain1.control->active_producers().version, 1u );
+
+   BOOST_TEST_REQUIRE(chain1.push_action(  eosio::chain::action({}, "addmember"_n, {}, participants_payload({"alice"_n, "bob"_n})), "addmember"_n.to_uint64_t() ) == "");
+   chain1.produce_blocks(11);
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 2);
+   chain1.produce_blocks(12);
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 0);
+   BOOST_CHECK(chain1.control->in_active_security_group(participants_t({"alice"_n, "bob"_n})));
+
+   BOOST_TEST_REQUIRE(chain1.push_action(  eosio::chain::action({}, "rmmember"_n, {}, participants_payload({"alice"_n})), "rmmember"_n.to_uint64_t() ) == "");
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 1);
+   chain1.produce_blocks(11+12);
```

Comment:
Add a test that adds/removes from the proposed security group and verifies that the proposed_security_group still has the changes and security group hasn't changed at LIB = last changed block - 1 and then is promoted at LIB = last changed block